crates/store/re_data_loader/src/loader_mcap.rs

```diff
@@ -102,129 +118,16 @@ impl DataLoader for McapLoader {
-            Arc::new(StringArray::from(vec![message_encoding.clone()])),
-        ))
-}
-
-fn from_schema(schema: &Arc<::mcap::Schema<'_>>) -> AnyValues {
-    use arrow::array::{StringArray, UInt16Array};
-
-    let ::mcap::Schema {
-        id,
-        name,
-        encoding,
-        data,
-    } = schema.as_ref();
-
-    // Adds a field of arbitrary data to this archetype.
-    AnyValues::new("rerun.mcap.Schema")
-        .with_field("id", Arc::new(UInt16Array::from(vec![*id])))
-        .with_field("name", Arc::new(StringArray::from(vec![name.clone()])))
-        .with_field("data", Arc::new(BinaryArray::from(vec![data.as_ref()])))
-        .with_field(
-            "encoding",
-            Arc::new(StringArray::from(vec![encoding.clone()])),
-        )
-}
-
-fn from_statistics(stats: &::mcap::records::Statistics) -> Result<AnyValues, ArrowError> {
-    let ::mcap::records::Statistics {
-        message_count,
-        schema_count,
-        channel_count,
-        attachment_count,
-        metadata_count,
-        chunk_count,
-        message_start_time,
-        message_end_time,
-        channel_message_counts,
-    } = stats;
-
-    let key_builder = UInt16Builder::new();
-    let val_builder = UInt64Builder::new();
-
-    let mut builder = MapBuilder::new(None, key_builder, val_builder);
-
-    for (&key, &val) in channel_message_counts {
-        builder.keys().append_value(key);
-        builder.values().append_value(val);
-        builder.append(true)?;
+        load_mcap(&contents, settings, &tx, None)
     }
-
-    let channel_message_counts = builder.finish();
-
-    Ok(AnyValues::new("rerun.mcap.Statistics")
-        .with_field(
-            "message_count",
-            Arc::new(UInt64Array::from_value(*message_count, 1)),
-        )
-        .with_field(
-            "schema_count",
-            Arc::new(UInt16Array::from_value(*schema_count, 1)),
-        )
-        .with_field(
-            "channel_count",
-            Arc::new(UInt32Array::from_value(*channel_count, 1)),
-        )
-        .with_field(
-            "attachment_count",
-            Arc::new(UInt32Array::from_value(*attachment_count, 1)),
-        )
-        .with_field(
-            "metadata_count",
-            Arc::new(UInt32Array::from_value(*metadata_count, 1)),
-        )
-        .with_field(
-            "chunk_count",
-            Arc::new(UInt32Array::from_value(*chunk_count, 1)),
-        )
-        .with_component::<components::Timestamp>(
-            "message_start_time",
-            vec![*message_start_time as i64],
-        )
-        .with_component::<components::Timestamp>("message_end_time", vec![*message_end_time as i64])
-        .with_field("channel_message_counts", Arc::new(channel_message_counts)))
 }
 
 #[cfg(not(target_arch = "wasm32"))]
 fn load_mcap_mmap(
     filepath: &std::path::PathBuf,
     settings: &DataLoaderSettings,
     tx: &Sender<LoadedData>,
+    layer_filters: Option<BTreeSet<String>>,
```

Comment:
This would make things more readable

```suggestion
    layer_filters: LayerFilter,
```

```rs
enum LayerFilter {
    All,
    Only(BTreeSet<LayerId>),
}
```